pkg/sql/rename_table.go

```diff
@@ -82,40 +83,50 @@ func (n *renameTableNode) ReadingOwnWrites() {}
 func (n *renameTableNode) startExec(params runParams) error {
 	p := params.p
 	ctx := params.ctx
-	oldTn := n.oldTn
-	newTn := n.newTn
 	tableDesc := n.tableDesc
 
-	oldUn := oldTn.ToUnresolvedObjectName()
-	prevDbDesc, prefix, err := p.ResolveUncachedDatabase(ctx, oldUn)
+	// Get the qualified name of the table.
+	oldTn, err := p.getQualifiedTableName(params.ctx, tableDesc.TableDesc())
 	if err != nil {
 		return err
 	}
-	oldTn.ObjectNamePrefix = prefix
+	prevDBID := tableDesc.ParentID
 
-	// Check if target database exists.
-	// We also look at uncached descriptors here.
-	newUn := newTn.ToUnresolvedObjectName()
-	targetDbDesc, prefix, err := p.ResolveUncachedDatabase(ctx, newUn)
-	if err != nil {
-		return err
+	var targetDbDesc *UncachedDatabaseDescriptor
+	// If the target new name has no qualifications, then assume that the table
+	// is intended to be renamed into the same database and schema.
+	newTn := n.newTn
+	if !newTn.ExplicitSchema && !newTn.ExplicitCatalog {
+		newTn.ObjectNamePrefix = oldTn.ObjectNamePrefix
+		var err error
+		targetDbDesc, err = p.ResolveUncachedDatabaseByName(ctx, string(oldTn.CatalogName), true /* required */)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Otherwise, resolve the new qualified name of the table. We are in the
+		// process of deprecating qualified rename targets, so issue a notice.
+		// TODO (rohany): Convert this to take in an unqualified name after 20.1 is released.
+		notice := pgnotice.Newf("database qualified RENAME targets are deprecated")
```

Comment:
nit: prefer to embed the notice to AppendNotice as one line:

```

		params.p.noticeSender.AppendNotice(
     pgnotice.Newf("database qualified RENAME targets are deprecated")
    )
```

also i think the message is a little confusing. maybe
```
errors.WithHint(
  pgnotice.Newf("renaming tables with a database or schema qualification is deprecated"),
 "Use SET SCHEMA (or whatever the database one is) instead",
)

Reply:
Good suggestion, updated. I'm not sure there is a way to move a table between databases in Postgres, so I'll just point to `SET SCHEMA` here.